spec/graphql/types/albums/album_search_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+module Types
+  module Albums
+    RSpec.describe QueryType, type: :request do
+      describe '.resolve' do
+        context 'when get a album by id' do
+          before do
+            album_search_request_success
+            photos_for_album_success
+            post '/graphql', params: { query: query_get_album(id: '1') }
+          end
+
+          it 'return the id album' do
```

Comment:
Instead of having a method that checks each field for album, you can do something similar with photos and build a hash and check against if if the response is what you were expecting.